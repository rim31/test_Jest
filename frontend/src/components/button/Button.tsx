import React from 'react';
import './button.css'

export default function Button(props: { label: string }) {
  return (
    <div data-testid="button" className="button-style">
      {props.label}
    </div>
  );
}
