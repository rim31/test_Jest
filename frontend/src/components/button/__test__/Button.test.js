import React from "react";
import ReactDOM from 'react-dom';
import Button from '../Button';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crash", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button></Button>, div);
})


// we are test a component in a DIV, passing a label should render the same text inside
it('render button correctly', () => {
  const { getByTestId } = render(<Button label="click me please"></Button>);
  // expect(getByTestId('button').toHaveTextContent("click me")); // OLD VERSION with // import "jest-dom/extend-expect"; https://github.com/testing-library/react-testing-library/issues/379
  expect(getByTestId('button').textContent).toBe('click me please')
})

it('render button correctly', () => {
  const { getByTestId } = render(<Button label="save"></Button>);
  expect(getByTestId('button').textContent).toBe('save')
})

// snapshot testing
it("match snapshot", () => {
  const tree = renderer.create(<Button label="save"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
})