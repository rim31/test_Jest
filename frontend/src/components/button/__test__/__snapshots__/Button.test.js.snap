// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  save
</div>
`;
